crates/interpreter/src/valid.rs

```diff
@@ -478,7 +476,7 @@ impl SideTable {
     }
 
     fn persist(self) -> MResult<Vec<SideTableEntry>, Check> {
-        self.entries.into_iter().map(|entry| SideTableEntry::new(entry.unwrap())).collect()
+        Ok(self.entries)
```

Comment:
Same here, we want to make sure we catch mistakes in debug mode.
```suggestion
        debug_assert!(self.entries.iter().all(|x| !x.is_invalid()));
        Ok(self.entries)
```

Reply:
Most probably, but it depends how we implement the "streaming" validation. Validation could take some abstract `Persist` structure as argument, which behaves like a `Vec<u8>` to which we can read and write. This way we actually directly write to flash and there's no copy.